f3/powertable.go

```diff
@@ -1,22 +1,81 @@
 package f3
 
-// A power table maps participant IDs to power values.
+import "sort"
+
+// PowerEntry represents a single entry in the PowerTable.
+// It includes an ActorID and its Weight
+type PowerEntry struct {
+	ID    ActorID
+	Power uint
+}
+
+// PowerTable maps ActorID to a unique index in the range [0, len(powerTable.Entries)).
+// Entries is the reverse mapping to a PowerEntry.
 type PowerTable struct {
-	Entries map[ActorID]uint
+	Entries []PowerEntry    // Slice to maintain the order
+	Lookup  map[ActorID]int // Map for quick index lookups
```

Comment:
Please be more clear about what this is indexing. "Maps actor ID to the index of the associated entry in Entries"

Reply:
Done.